gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -336,32 +338,33 @@ impl ClnHtlcInterceptor {
             {
                 Ok(_) => {
                     // Open a channel to receive the outcome of the HTLC processing
-                    let (sender, receiver) = oneshot::channel::<Action>();
+                    // Use BOLT 4 failure message in case of error. See
+                    // https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages
+
+                    let (sender, receiver) = oneshot::channel::<serde_json::Value>();
                     self.outcomes.lock().await.insert(short_channel_id, sender);
 
                     // If the gateway does not respond within the HTLC expiry,
                     // Automatically respond with a failure message.
-                    return tokio::time::timeout(
-                        Duration::from_secs(5),
-                        self.await_htlc_processing(receiver, payment_hash.clone()),
-                    )
+                    return tokio::time::timeout(Duration::from_secs(5), async {
```

Comment:
This should probably be more than 5s. It would be totally fine even if it is 1h in most cases. I'd go for 30s or so as a worst case. The GW needs to buy the preimage (1 epoch) and the federation needs to decrypt the preimage (1+ up to f epochs).

We should also think about what to do when shut down because CLN shuts down. Can we subscribe to the appropriate signals and cancel all HTLC in that case? Idk how CLN shuts down, best to speak to the CLN team.

Reply:
Ack. Marking this for quick [follow ups](https://github.com/fedimint/fedimint/pull/1224#issuecomment-1380933799). We need to do several iterations on this base implementation